kapitan/inputs/helm/template.go

```diff
@@ -0,0 +1,320 @@
+		if !ok {
+			dest[k] = v
+			continue
+		}
+		// Edge case: If the key exists in the destination, but isn't a map
+		destMap, isMap := dest[k].(map[string]interface{})
+		// If the source map has a map for this key, prefer it
+		if !isMap {
+			dest[k] = v
+			continue
+		}
+		// If we got to this point, it is a map in both, so merge them
+		dest[k] = mergeValues(destMap, nextMap)
+	}
+	return dest
+}
+
+// vals merges values from files specified via -f/--values and
+// directly via --set or --set-string or --set-file, marshaling them to YAML
+func vals(valueFiles valueFiles, values []string, stringValues []string, fileValues []string) ([]byte, error) {
+	base := map[string]interface{}{}
+
+	// User specified a values files via -f/--values
+	for _, filePath := range valueFiles {
+		currentMap := map[string]interface{}{}
+
+		var bytes []byte
+		var err error
+		if strings.TrimSpace(filePath) == "-" {
+			bytes, err = ioutil.ReadAll(os.Stdin)
+		} else {
+			bytes, err = readFile(filePath)
+		}
+
+		if err != nil {
+			return []byte{}, err
+		}
+
+		if err := yaml.Unmarshal(bytes, &currentMap); err != nil {
+			return []byte{}, fmt.Errorf("failed to parse %s: %s", filePath, err)
+		}
+		// Merge with the previous map
+		base = mergeValues(base, currentMap)
+	}
+
+	// User specified a value via --set
+	for _, value := range values {
+		if err := strvals.ParseInto(value, base); err != nil {
+			return []byte{}, fmt.Errorf("failed parsing --set data: %s", err)
+		}
+	}
+
+	// User specified a value via --set-string
+	for _, value := range stringValues {
+		if err := strvals.ParseIntoString(value, base); err != nil {
+			return []byte{}, fmt.Errorf("failed parsing --set-string data: %s", err)
+		}
+	}
+
+	// User specified a value via --set-file
+	for _, value := range fileValues {
+		reader := func(rs []rune) (interface{}, error) {
+			bytes, err := readFile(string(rs))
+			return string(bytes), err
+		}
+		if err := strvals.ParseIntoFile(value, base, reader); err != nil {
+			return []byte{}, fmt.Errorf("failed parsing --set-file data: %s", err)
+		}
+	}
+
+	return yaml.Marshal(base)
+}
+
+// generates name using Sprig template
+func generateName(nameTemplate string) (string, error) {
+	t, err := template.New("name-template").Funcs(sprig.TxtFuncMap()).Parse(nameTemplate)
+	if err != nil {
+		return "", err
+	}
+	var b bytes.Buffer
+	err = t.Execute(&b, nil)
+	if err != nil {
+		return "", err
+	}
+	return b.String(), nil
+}
+
+//export renderChart
+func renderChart(c_chartpath, c_outputDir, c_valueFile *C.char) *C.char {
+	chartPath := C.GoString(c_chartpath)
+	outputDir := C.GoString(c_outputDir)
+	valueFile := C.GoString(c_valueFile)
+	// values in YAML file
+	var valueFiles []string
+	if valueFile != "" {
+		valueFiles = append(valueFiles, valueFile)
+	}
+	var values []string
+	// to force string values
+	var stringValues []string
+	var fileValues []string
+	nameTemplate := ""
+	releaseName := ""
```

Comment:
@yoshi-1224 we should also allow passing this value through the inventory.
Maybe by implementing `helm_params` (similar to `helm_values`) where one can set params like `namespace`, `release`, etc... ?